research/global-config/index.md

```diff
@@ -0,0 +1,361 @@
+- Configuration migrations are straightforward for the service team; it is just a change in a file that belongs to the team.
+- Highest performance due to direct configuration access; no need to load and pre-process other files.
+- Quickest implementation; this approach is the same as for the packit-service configuration file.
+
+## Disadvantages
+
+- It is probably easier, for the final user, to look for the service configuration file in its own repo instead of a service repo.
+- Less user engagement due to limited visibility of configuration changes. Even though a configuration migration can be simpler for the service team it could be less explicit for the final user. Being able to change behaviour without the user acknowledging it could not be a good idea. In packit there already is a configuration migration script.
+- The global configuration file could really grow huge and be difficult to maintain both for the service team and for a final user that wants to contribute to it.
+- Lacks ecosystem-specific configuration management. There is no encapsulation for "middle layer" knowledge, no easy way to manage ecosystem configurations. Configurations could be grouped in different nodes with different defaults but there is no easy way, for the service team, to know if the user who is asking to update a configuration has the rights for doing so.
+- Differs from Packit's current approach, potentially confusing users. This could work mainly for the _Fedora CI_ and could confuse users that use the standard packit configuration for _upstream continuous integration_ and _release synchronization_.
+
+# Bottom-Up solution
+
+Project-side configuration with ability to reference global configurations.
+
+## Advantages
+
+- User-friendly and explicit. Everything is on the user side (more quick for him) and the file will never grow too much because of other project details.
+- Flexible customization through layered configurations and better ecosystem-specific configuration support.
+- Consistent with current Packit implementation.
+
+## Disadvantages
+
+- More complex configuration migrations because they are on the user side.
+- Additional processing overhead. It will require time to load multiple configuration layers.
+- Requires implementation of inheritance/templating mechanisms.
+
+## Configuration Layers
+
+### Service-Side Layer
+
+The last layer of a configuration chain could be on the service side and it could be referenced in a sort of opt-in mechanism or it could be automatically applied.
+
+### User-Side Layer
+
+At the moment the packit service needs just one configuration file on the user project side both for the **upstream continuous integration** and for the **release synchronization**.
+When implementing the **Fedora CI** (which is a _downstream continuous integration_) a new configuration file could be required or the configuration can be merged with the existing one.
```

Comment:
We just need to be careful not to introduce too many file names because of the forge API issues (but Forgejo might handle these more easily).